### PR TITLE
feature: Augment /learn response with Azure Resource Graph query results

### PR DESCRIPTION
**Pull request by JasonYeMSFT**
Also moved the learn slash command implementation to its own directory, like other slash commands.

### CODE REVIEW

**Comment by JasonYeMSFT** [comment]
This breaks the generic case where no ARG query should be done. The language model reacts to generic terms like "Azure Storage" and the ARG API hallucinates by generating a query that doesn't return useful information.

**Comment by MRayermannMSFT** [comment]
You should rebase with origin/main

File: src/chat/learn/learnSlashCommand.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+                    argQueryResult = await queryAzureResourceGraph(actionContext, resourceInfo, request);
+                    ext.outputChannel.debug("argQueryResult", argQueryResult);
+                    if (argQueryResult) {
+                        finalResponse = await augmentLearnResponse(learnResponse, JSON.stringify(argQueryResult.response), request);
+                    }
+                }
+                ext.outputChannel.debug("finalResponse", finalResponse);
+                request.responseStream.markdown(finalResponse + "\n");
+                if (argQueryResult) {
+                    request.responseStream.markdown(`The answer is generated based on the result of an Azure Resource Graph query.`);
```
Comment (MRayermannMSFT) [inline src/chat/learn/learnSlashCommand.ts:76]: "This content"

File: src/chat/agent.ts [hunk 2]
```diff
@@ -13,8 +13,8 @@ import { agentHelpCommandName, getAgentHelpCommand } from "./agentHelpSlashComma
 import { argQueryCommand } from "./argQuery/argQuerySlashCommand";
 import { AgentBenchmarker } from "./benchmarking/AgentBenchmarker";
 import { argQueryBenchmarks, defaultBenchmarks, helpBenchmarks, multiPromptBenchmarks } from "./benchmarking/agentBenchmarks";
-import { getLearnCommand } from "./commonCommandsAndHandlers";
 import { appServiceExtensionSlashCommandsOwner, azdExtensionSlashCommandsOwner, containerAppsExtensionSlashCommandsOwner, databasesExtensionCosmosDbSlashCommandsOwner, databasesExtensionPostgreSQLSlashCommandsOwner, functionsExtensionSlashCommandsOwner, staticWebAppsExtensionSlashCommandsOwner, storageExtensionSlashCommandsOwner, virtualMachinesExtensionSlashCommandsOwner } from "./extensions/KnownAzureExtensions";
+import { getLearnCommand } from "./learn/learnSlashCommand";
```
Comment (MRayermannMSFT) [inline src/chat/agent.ts:17]: Let's update the topics for the agent's learn command to be something like:

"Azure, coding for Azure, or your Azure resources"

**Review by MRayermannMSFT — APPROVED** [review]

**Pull request merged**